### ISSUE TITLE
Response Message from a Failed `validate()` Should be Changed

### ISSUE THREAD

**Issue by vic-ma** [rh-jmc-team/container-jfr#195]
After #193, it no longer makes sense for the response message from a failed `validate()` call to only talk about improper arguments, since the `validate()` in `UploadRecordingCommand` can now fail due to an improper environment variable.

**Comment by andrewazores** [comment]
Just my initial idea - rather than returning a boolean, perhaps the `validate` method (in general) should have a void return and instead throw an exception when validation fails. Then the command executor can catch that exception and handle it accordingly, either by printing the exception message to the `ClientWriter` or by putting the exception message into a failed serialized output.

**Comment by vic-ma** [comment]
In `BatchModeExecutorTest`, some of the tests (all the ones from [this one](https://github.com/rh-jmc-team/container-jfr/blob/main/src/test/java/com/redhat/rhjmc/containerjfr/tui/tty/BatchModeExecutorTest.java#L174) and down) use `connect` and `disconnect` as example commands. I imagine that these are just holdovers from before the connectionless API changes and should be replaced with existing commands? And it's probably substantial enough to warrant its own PR?

**Comment by andrewazores** [comment]
Yea, those are all holdovers from before. I don't think it necessarily needs to be changed however - all of those command names are completely arbitrary, since they're being passed to a mock command registry for mock validation and execution. All of those examples were written using real command names at the time, but there's no real reason that that needs to be the case.

If you want to go ahead and change them for illustrative purposes then by all means feel free. It could be a separate PR or just a separate, final commit containing only those changes tacked on to the end of the PR that comes out of this issue, since it's cleanup in the same tests that will already be modified here.

**Comment by vic-ma** [comment]
I guess since it's not much work and is a little beneficial, I do think it's worth just renaming the commands. And I'll just do it in a final commit like you said.

**Comment by vic-ma** [comment]
Ah, just realized that my change replaced all the `cw.println()` in the commands' validate methods with exception throws, so there aren't any backend logs for failed validations anymore. If I add them back, it would be good for serialized execution, since then there will be a json response that includes the failed validation message, and there will also be the same message in the backend logs, which is similar to how it is in upstream.

But for the interactive execution, adding the `cw.println()` back would cause the same message to be printed twice in the terminal: once coming from the `validate()` method in the command, due to `cw.println()`; and once coming from the `validateCommands()` method in `InteractiveShellExecutor`, due to printing the exception's message.

So right now, in upstream, a failed validation in interactive mode looks something like this:
```
> list foo bar
Expected one argument: hostname:port, ip:port, or JMX service URL    <-- from validate() in ListCommand
	"[foo, bar]" are invalid arguments to list                   <-- from validateCommands() in InteractiveShellExecutor
```
And my suggestion would be to just keep the `cw.println()` in the commands' `validate()` method, since removing it would mean that serialized execution needs some workaround to get backend logs. And then change `validateCommands()` to not print the exception's message at all, but just something like `Command \"%s\" could not be validated` if the exception is caught. So it would look something like this:
```
> list foo bar
Expected one argument: hostname:port, ip:port, or JMX service URL
	Command "list" could not be validated
```

This does mean that the structure of the tty and websocket executors would become a bit asymmetrical, since one would use the exception's message in its response, and the other just ignores it.

What do you think about this, or do you have any alternative suggestions? [Here](https://github.com/rh-jmc-team/container-jfr/compare/main...vic-ma:change-validate-response-msg) is my PR's current changes if you're interested.

**Comment by andrewazores** [comment]
I think your suggestion sounds fine. The priority is always on the serialized form of the commands and any executors that use them (so, just the WebSocket one, at least for now). Your example of what an interactive mode validation failure would look like seems completely fine by me. At some point the interactive mode of commands and executor should probably be removed anyway, or at least refactored heavily to work top of the serialized forms, rather than being a parallel code path.

**Comment by vic-ma** [comment]
In testing files, how do we decide when to statically import test methods, versus just importing their class and accessing using `Class.method()`? For example, for things like `MatcherAssert.assertThat()` or `Assertions.assertThrows()`.

**Comment by andrewazores** [comment]
Just a style or preference thing. I used static imports when I was using an editor where the code completion made that easier. Now that I've changed back to my old editor (out of habit), it's a bit more annoying to do static imports because I have to go to the top and edit the import manually, so I just don't generally bother...
